### DISCUSSION TITLE
Attribute Ranking Rules

### DISCUSSION THREAD

**Discussion by ThisIsMyFavouriteJar** [meilisearch/product#549] [Feedback & Feature Proposal]
Hello,

Very happy with Meilisearch so far! When looking at the attribute ranking rules in your documentation it says that 

> ... attributes containing the query words at the beginning of the attribute will be considered more relevant than documents containing the query words at the end of the attributes.

[From this part of the docs](https://docs.meilisearch.com/learn/core_concepts/relevancy.html#_5-sort)

This makes sense for attributes with a lot of text, but for smaller attributes makes less sense. If I search a product "Speaker" I want to score the product names "JBL Speaker" and "Speaker cable" as equal, and then determine their actual ranking based on subsequent attributes. (Speaker Cable always scores higher in this case, due to the term appearing right at the start.)

Would it make sense to make the position of a word not important for certain attributes? 

Happy to hear your thoughts or if I am doing something wrong!

**Comment by curquiza** [comment]
Hello @ThisIsMyFavouriteJar thanks for this feedback, I moved this discussion into [the product repository](https://github.com/meilisearch/product/discussions) where we gather feedback

I let @gmourier answer if he needs any precision

**Comment by gmourier** [comment]
Hi @ThisIsMyFavouriteJar 👋

Have you tried to set a specific field ordering in the `searchableAttributes`? e.g. `["name", "desc", ...]`. If other attributes contain `speaker` it may re-rerank the results and gives you the desired results.

Thanks
